magma/lib/magma/attributes/foreign_key.rb

```diff
@@ -6,6 +6,10 @@ def initial_column_name
       foreign_id
     end
 
+    def revision_to_loader(record_name, new_value)
+      [ name, new_value&.to_s ]
+    end
+
```

Comment:
if there is a foreign key in the revision (not nil), cast it to a string.

Reply:
Should internal be `name` or `record_name`?
Below as well.

Reply:
this is correct, the record_name is used by e.g. FileCollectionAttribute#revision_to_loader to set filenames and such, not used here.